lexicon/providers/safedns.py

```diff
@@ -0,0 +1,200 @@
+
+        self._patch(
+            '/zones/{0}/records/{1}'.format(self.domain, identifier), data)
+
+        LOGGER.debug('update_record: %s', True)
+        return True
+
+    def _delete_record(self, identifier=None, rtype=None, name=None, content=None):
+
+        delete_record_ids = []
+
+        # If we've not been given an identifier, search for matching records.
+        # NOTE, this could cause multiple records to be removed.
+        if not identifier:
+            records = self._list_records(rtype, name, content)
+            delete_record_ids = [record['id'] for record in records]
+        else:
+            delete_record_ids.append(identifier)
+
+        LOGGER.debug('delete_records: %s', delete_record_ids)
+
+        for delete_record_id in delete_record_ids:
+            self._delete(
+                '/zones/{0}/records/{1}'.format(self.domain, delete_record_id))
+
+        LOGGER.debug('delete_record: %s', True)
+        return True
+
+    # As there is no _patch method available from Provider, let's create our own
+    # by creating a _request with action 'PATCH'.
+    def _patch(self, url='/', data=None, query_params=None):
+        return self._request('PATCH', url, data=data, query_params=query_params)
+
+    def _request(self, action='GET', url='/', data=None, query_params=None):
+        if data is None:
+            data = {}
+        if query_params is None:
+            query_params = {}
+        default_headers = {
+            'Accept': 'application/json',
+            'Content-Type': 'application/json',
+            'Authorization': '{0}'.format(self._get_provider_option('auth_token'))
+        }
+        if not url.startswith(self.api_endpoint):
+            url = self.api_endpoint + url
+
+        response = requests.request(action, url, params=query_params,
+                                    data=json.dumps(data),
+                                    headers=default_headers)
+
+        # Sort this out to work properly, writing errors to the correct place
+        if 'content' in response:
+            resp = json.loads(response.content)
+            if 'errors' in resp:
+                for error in resp['errors']:
+                    print("ERROR: " + error['detail'])
+
+        # If the request fails for any reason, throw an error.
+        response.raise_for_status()
+
+        # There is no JSON returned when calling with DELETE
+        if action != 'DELETE':
+            return response.json()
+
+        return True
+
+    # The content of TXT entries must be quoted. This static method ensures that.
+    @staticmethod
+    def _add_quotes(rtype, content):
+        if rtype == 'TXT':
+            if not content.startswith('"') and not content.endswith('"'):
+                return '"{0}"'.format(content)
+        return content
+
+    # When we're getting a response, we don't want quotes to wrap the content.
+    # This method removes them
+    @staticmethod
+    def _remove_quotes(rtype, content):
```

Comment:
Maybe you can reuse `_clean_TXT_record` from the base provider instead ?

Reply:
I didn't spot that in the base class. Definitely!